draft-ietf-tls-esni.md

```diff
@@ -438,10 +426,24 @@ echo_hrr_key = context.Export("tls13-echo-hrr-key", 16)
 ~~~
 
 Note that the HPKE algorithm identifiers are those which match the client's
-chosen Ciphersuite, according to {{hpke-map}}. The client MAY replace any large,
+chosen CipherSuite, according to {{hpke-map}}. The client MAY replace any large,
 duplicated extensions in ClientHelloInner with the corresponding "outer_extensions"
 extension, as described in {{outer-extensions}}.
 
+The client then generates a ClientHelloInner value. In addition to the normal
+values, ClientHelloInner MUST also contain:
+
+ - an "echo_nonce" extension
+ - TLS padding {{!RFC7685}}
+
+The padding SHOULD contain X bytes of padding, where X = ECHOConfig.maximum_name_length - length(dns_name),
+rounded up to the nearest multiple of 16, and dns_name is the DNS name in the
+ClientHelloInner "server_name" extension.
```

Comment:
Not sure how best to capture this in text... if the client also, say, varies the ALPN advertisement by server, it may want to add padding as needed to mask that variation. That'll depend on how the client varies ALPN.

Padding is such a mess. :-(

Reply:
Yeah, that's why I suggest we punt on this issue for now. We either need to pad across the entire CH (which was what the previous maximum_message_length value did), or accept some leakage.